### PR TITLE
Add functionality to deploy custom neofs environments

### PR DESCRIPTION
**Pull request by evgeniiz321**
Closes https://github.com/nspcc-dev/neofs-testcases/issues/672

I've tried to remove as much code as possible to ease the review. In this PR:
- single test to verify deployment of a 'simple' env with 1 IR 1 SN 1 S3GW and show how to use created classes
- base class for NeoFSEnv and separate classes for IR, SN, S3GW

Everything should be self descriptive without additional comments, if not - leave a comment.

### CODE REVIEW

File: pytest_tests/test_env.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+def wallet() -> NodeWallet:
+    wallet_name = f"{str(uuid.uuid4())}.json"
+    wallet_path = os.path.join(os.getcwd(), wallet_name)
+    wallet_password = "password"
+    wallet_address = init_wallet(wallet_path, wallet_password)
+    return NodeWallet(path=wallet_path, address=wallet_address, password=wallet_password)
+
+
+@pytest.fixture
+def s3_creds(neofs_env: NeoFSEnv, zero_fee, wallet: NodeWallet) -> tuple:
```
Comment (roman-khimov) [inline pytest_tests/test_env.py:52]: S3 as a part of base env?

File: src/neofs_testlib/env/env.py [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+        # utilities
+        self.neofs_adm_path = os.getenv("NEOFS_ADM_BIN", "./neofs-adm")
+        self.neofs_cli_path = os.getenv("NEOFS_CLI_BIN", "./neofs-cli")
+        self.neo_go_path = os.getenv("NEO_GO_BIN", "./neo-go")
+        self.neofs_ir_path = os.getenv("NEOFS_IR_BIN", "./neofs-ir")
+        self.neofs_node_path = os.getenv("NEOFS_NODE_BIN", "./neofs-node")
+        self.neofs_s3_authmate_path = os.getenv("NEOFS_S3_AUTHMATE_BIN", "./neofs-s3-authmate")
+        self.neofs_s3_gw_path = os.getenv("NEOFS_S3_GW_BIN", "./neofs-s3-gw")
+        self.neofs_rest_gw_path = os.getenv("NEOFS_REST_GW_BIN", "./neofs-rest-gw")
+        self.neofs_http_gw_path = os.getenv("NEOFS_HTTP_GW_BIN", "./neofs-http-gw")
```
Comment (roman-khimov) [inline src/neofs_testlib/env/env.py:52]: I'd better have gateways defined separately somewhere, many tests don't care about them at all. Those who care, should just run an instance as needed.

Comment (evgeniiz321) [reply src/neofs_testlib/env/env.py:52]: This is not a definition of gateways, it is just the way to organize env vars in one place. If tests don't care - they won't use it, there is no requirement to specify any of these env vars. Gateways are indeed separate classes (you can see the class for s3 gateway in this PR). I can move related paths there, but it seems to be more convenient to have them in one place.

File: pytest_tests/test_env.py (hunk 1 above)
Comment (evgeniiz321) [reply pytest_tests/test_env.py:52]: In this particular case - yes. But actually any kind of env is configured via this kind of classmethods or fixtures:

```
  @classmethod
    def simple(cls) -> "NeoFSEnv":
        neofs_env = NeoFSEnv()
        neofs_env.deploy_inner_ring_node()
        neofs_env.deploy_storage_node()
        neofs_env.deploy_s3_gw()
        return neofs_env
```

So we can have whatever we want. But for the purpose of this PR - to have a working example - I use s3 gate here.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**